describealign.py

```diff
@@ -1125,7 +1126,7 @@ def combine_gui(video_files, audio_files, config_path):
   combine_window.close()
 
 def main_gui():
-  config_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config.ini')
+  config_path = platformdirs.user_config_path(appname='describealign', ensure_exists=True) / 'config.ini'
```

Comment:
I can imagine the old method here having problems, but I don't think I ever saw any issues even with the Mac binaries.  Where were you seeing a problem with writing the config?  If we are gonna change this, I'd rather not break existing installs by resetting their configs when they upgrade with pip.  But that could be solved by checking the old config location and moving it over if there isn't a new one already.

Reply:
On mac, it would attempt to write the config in the installed app dir /Applications/describealign.app which is owned by root. We can definitely add checks for the old location yeah

Reply:
Added migration function to move the old config